### ISSUE TITLE
1.7.237 - Logfile provider - CSV filetype empty

### ISSUE THREAD

**Issue by jpedlow** [PowershellFrameworkCollective/psframework#539]
Hello!

I've been attempting to get PSF going, but am having issues saving as CSV. 

Here's my demo code, note that I'm using double-quotes and have set the path to lead with PSSCRIPTROOT:

#logfile setup (PSFramwork Module)
$paramSetPSFLoggingProvider = @{
    Name         = "logfile"
    InstanceName = "<taskname>"
    FilePath     = "$psscriptroot\TaskName-%Date%.csv"
    Enabled      = $true
}
Set-PSFLoggingProvider @paramSetPSFLoggingProvider

foreach ($i in 1..10) {
    Write-PSFMessage -Level Warning -Message "Error: TEST ERROR $i"
    Start-Sleep -Milliseconds (Get-Random -Min 500 -Max 1000)
}

Issues:
After script completion, the CSV will be:
- empty
- read-only locked (until the terminal process is closed)

Things that **do** work:
- CMTRACE, JSON, XML (MUST be running PS Elevated to Administrator)

Also worth noting, is that the Files are held LOCKED until the originating PowerShell process is closed. 
Or said differently, even though script completion has occurred, I cannot delete/modify the file until PowerShell has been closed.

Thanks very much! :)

**Comment by FriedrichWeinmann** [comment]
Hi @jpedlow,
thanks for raising this and - especially - the incredibly useful pointer at CSV being the special case.
Found the problem and will ship a fix today.

That said, I cannot find a scenario requiring elevation - can you detail just what requires that for you, an example and environment details?
The file being locked was necessary, as the logging had faced some issues with antivirus interfering with the logging and also for performance reasons.
To release a logfile, disable the logging provider, for example like this:

```powershell
# Ensure all messages are processed
Wait-PSFMessage

# Disable Provider
Set-PSFLoggingProvider -Name logfile -Enabled $false
```

Since the logging provider is running in the background, it has no way of knowing when your script is over, or that it should only handle messages from one single script, so you need to tell it its job is done :)
Unless of course the console closes once the script is done (e.g., during a scheduled task), in which case it's redundant.

**Comment by FriedrichWeinmann** [comment]
Released/Resolved in 1.7.244
